Source/SwiftLintFramework/Rules/Idiomatic/RedundantVoidReturnRule.swift

```diff
@@ -91,4 +92,16 @@ public struct RedundantVoidReturnRule: ConfigurationProviderRule, SubstitutionCo
     public func substitution(for violationRange: NSRange, in file: File) -> (NSRange, String) {
         return (violationRange, "")
     }
+
+    private func containsVoidReturnTypeBasedOnTypeName(dictionary: [String: SourceKitRepresentable]) -> Bool {
+        guard SwiftVersion.current >= .fourDotOne else {
+            return true
```

Comment:
This doesn't seem to match what the function name says.

Reply:
Are you referring to this first `guard/else/return`? This is because this information is only available in SourceKit after Swift 4.1 and I wanted to keep the rule working as-is on older Swift versions.

Reply:
We should short circuit this earlier. This doesn’t belong in this function.

Reply:
Sure, I can change the call site but I thought it'd add more complexity:

```swift
guard functionKinds.contains(kind),
   SwiftVersion.current < .fourDotOne || containsVoidReturnTypeBasedOnTypeName(dictionary: dictionary),
// ... else {

}
```

Reply:
Why would you run this check `functionKinds.contains(kind)` when `SwiftVersion.current < .fourDotOne` will either always fail or always pass?

Reply:
Because `containsVoidReturnTypeBasedOnTypeName` is really just a shortcut for catching negatives faster on Swift 4.1+. We still want to support all Swift versions in this rule.

Reply:
I see. My main concern for this change then is that we return a value that doesn't answer the question asked by the function name:

Q: contains void return type based on type name?
A: Always yes for swift < 4.1, regardless of if the structure contains void return type based on type name

Maybe rename to something that better reflects what the function does?

Reply:
That's a good point. I wasn't too concerned about this because it's a private function, but I've  added an auxiliar function, hopeful that will make things more explicit 🙌